### PR TITLE
Set itemType on Server

### LINKED ISSUE #783: Collections: Set item type explicitly on the server
## Problem

When you load an item type collection page, it initially says "Macbook Parts" and then flashes to "Macbook Batteries". We thought upgrading instantsearch would have resolved this, so we dropped the feature that parsed the url on the server and set the item type in the productlist prop.

## Do This

When on the server, set `productlist.itemType` if there is an item type in the url. Use this instead of `devicePartsItemType` in the hook.

CC @sterlinghirsh

### PR DESCRIPTION
**Pull request by sumukhipandey**
This PR is basically a revert of this commit https://github.com/iFixit/react-commerce/commit/5155bbcca4ecfc6e7d287e91f7ca53bf10c3ba70 We got rid of this feature and upgraded to instantSearch, but this was causing a weird loading pattern.

Closes #783 

CC: @masonmcelvain

### CODE REVIEW

File: frontend/components/common/InstantSearchProvider.tsx [hunk 1]
```diff
@@ -130,6 +130,7 @@ export function InstantSearchProvider({
                .filter((part) => part !== '');
             const partsOrTools = pathParts.length >= 1 ? pathParts[0] : '';
             const deviceHandle = pathParts.length >= 2 ? pathParts[1] : '';
+            const itemTypeHandle = pathParts.length >= 3 ? pathParts[2] : '';
```
Comment (masonmcelvain) [inline frontend/components/common/InstantSearchProvider.tsx:133]: This var doesn't get used anywhere, so I don't think we need it

**Review by masonmcelvain — CHANGES_REQUESTED** [review]
CR ✌🏻 Nice work @sumukhipandey! This adds `productList.deviceItemType` on the server, but doesn't yet use it anywhere. We need to add it in the hook (like in this commit https://github.com/iFixit/react-commerce/commit/a5481b5c597546946dd30bb5c24bf05bdfcdae9b) so that the app can use it on the server.

**Comment by masonmcelvain** [comment]
forgot to dev_block ✌🏻 on that

**Review by masonmcelvain — DISMISSED** [review]
CR ✌🏻 I think this is right, nice work!

**Comment by masonmcelvain** [comment]
un_dev_block ✌🏻

**Comment by erinemay** [comment]
dev_block 🌱 
The breadcrumbs got a smidge messed up:
image
https://react-commerce-git-set-item-type-on-server-ifixit.vercel.app/Parts/Mac/Bluetooth_Board

Note: It looks like what you're trying to do worked. There's an intermediate point during page load that shows the item type name is getting loaded instead of "Parts". It just finishes loading and looks like it does in that first image ☝️ 
image

I attached a couple of screen recordings with the speed toggled way down so you can see how the page loads:

<details><summary>Branch</summary>

https://user-images.githubusercontent.com/72166715/201189058-88886c25-71cd-483d-ac67-0423b32d81f6.mp4

</details>

<details><summary>Main</summary>

https://user-images.githubusercontent.com/72166715/201189072-9e1b7697-d8d9-4e15-9d72-96223de03a51.mp4

</details>

**Comment by sumukhipandey** [comment]
> dev_block 🌱 The breadcrumbs got a smidge messed up: image https://react-commerce-git-set-item-type-on-server-ifixit.vercel.app/Parts/Mac/Bluetooth_Board
> 
> Note: It looks like what you're trying to do worked. There's an intermediate point during page load that shows the item type name is getting loaded instead of "Parts". It just finishes loading and looks like it does in that first image ☝️ image
> 
> I attached a couple of screen recordings with the speed toggled way down so you can see how the page loads:
> 
> Branch
> Main

Thanks Erin! I am not sure what's causing the breadcrumbs bug here... @masonmcelvain do you have any idea as to why this might've happened?

**Comment by erinemay** [comment]
I've seen this exact issue pop up on a pull semi-recently. I'm gonna hunt for it.

**Comment by masonmcelvain** [comment]
Thank you @erinemay! I knew we had seen this before. I'm also looking into it.

**Comment by masonmcelvain** [comment]
un_dev_block ✌🏻 Super strange behavior, I can't explain why the breadcrumbs were borked. Regardless, we solved the problem before by not using the SSR hack for the breadcrumbs, so let's reuse that solution. (@danielbeardsley)

**Comment by k0rvusk0r4x** [comment]
- Breadcrumbs are no longer messed up.
- I am not seeing the incorrect loading behavior.

QA 😺

File: frontend/templates/product-list/sections/FilterableProductsSection/useDevicePartsItemType.tsx [hunk 2]
```diff
@@ -17,6 +18,9 @@ export function useDevicePartsItemType<T extends ProductListAttributes>(
       // `Item Type` is a single select, so just use the first value if it exists.
       return itemTypeRefinement?.refinements[0]?.value;
    }, [items]);
+   if (typeof window === 'undefined' && productList.deviceItemType) {
```
Comment (danielbeardsley) [inline frontend/templates/product-list/sections/FilterableProductsSection/useDevicePartsItemType.tsx:21]: I kinds feel like this whole routine is somewhat convoluted. I'm gonna try something.

Comment (danielbeardsley) [reply frontend/templates/product-list/sections/FilterableProductsSection/useDevicePartsItemType.tsx]: The pushed commit seems to not create the bookmarks weirdness and I feel it's clearer to read.

**Comment by danielbeardsley** [comment]
CR :+1:

**Review by danielbeardsley — APPROVED** [review]
CR :+1:

**Comment by k0rvusk0r4x** [comment]
Continues to function the same as [here](https://github.com/iFixit/react-commerce/pull/998#issuecomment-1312030669).

QA 😺

**Pull request merged**